### PR TITLE
Release dredd 12.2.0

### PR DESCRIPTION
**Pull request by kylef**
- Updated fury toolchain (https://github.com/apiaryio/dredd/pull/1623)
- Added support for JSON Schema Draft 6 and 7 (https://github.com/apiaryio/dredd/pull/1622)

### Post Merge Release:

- [ ] `npx lerna publish from-package`
- [ ] `git tag -a -m dredd@12.2.0 dredd@12.2.0`
- [ ] `git tag -a -m dredd-transactions@9.1.0 dredd-transactions@9.1.0`
- [ ] `git push --tags`

### CODE REVIEW

**Comment by artem-zakharchenko** [comment]
Shouldn't we have Dredd automatically published by the Apiary bot once it detects a version bump commit? I thought that automation part remains.

File: packages/dredd/scripts/smoke.sh [hunk 1]
```diff
@@ -2,52 +2,46 @@
...
+  mv "$PROJECT_DIR/packages/$PACKAGE/$TARBALL" "$TMPDIR"
+
+  cd "$TMPDIR" && npm install --no-save "$TARBALL"
+}
+
+git clone https://github.com/apiaryio/dredd-example.git "$TMPDIR"
+install dredd-transactions
+install dredd
 
-if [ ! -z "$CIRCLECI" ]; then
```
Comment (kylef) [inline packages/dredd/scripts/smoke.sh]: I've removed the requirement that smoke tests are only able to be ran on CircleCI. It made it harder to test and I didn't see the need to disallow running this locally.

File: packages/dredd/scripts/smoke.sh [hunk 2]
```diff
@@ -2,52 +2,46 @@
...
-
-  # Test the JavaScript API
-  node -e 'process.exitCode = (new (require("dredd"))({})).run ? 0 : 1;'
+cd "$TMPDIR"
+
+# Install dredd-example's dependencies
+npm install
+
+# Assert that Protagonist (the C++ dependency) was not installed
+if [[ "$(find node_modules -name protagonist)" != "" ]]; then
```
Comment (kylef) [inline packages/dredd/scripts/smoke.sh]: This has been verified by replacing "protagonist" with "drafter.js" and watching it fail.

**Comment by kylef** [comment]
> Shouldn't we have Dredd automatically published by the Apiary bot once it detects a version bump commit? I thought that automation part remains.

Yeah, we should automate the process. This was a one-off manual release until automatic or semantic release is setup.

File: packages/dredd/scripts/smoke.sh (hunk 1 above)
Comment (artem-zakharchenko) [reply packages/dredd/scripts/smoke.sh]: I think the historical reason to forbid that is due to smoke test performs file-system operations that may break something, if run on local. However, during my debugging of the smoke test there hasn't been such occasions, so I support removing this check.

Comment (kylef) [reply packages/dredd/scripts/smoke.sh]: Yeah, and I've moved to using a tempoary directory so we shouldn't leave any  state around anymore.

**Comment by kylef** [comment]
```shell
$ npx lerna publish from-package
lerna notice cli v3.18.3
lerna info versioning independent

Found 2 packages to publish:
 - dredd-transactions => 9.1.0
 - dredd => 12.2.0

? Are you sure you want to publish these packages? Yes
lerna info publish Publishing packages to npm...
lerna notice Skipping all user and access validation due to third-party registry
lerna notice Make sure you're authenticated properly ¯\_(ツ)_/¯
lerna info lifecycle dredd-transactions@9.1.0~prepack: dredd-transactions@9.1.0

> dredd-transactions@9.1.0 prepack /Users/kyle/Projects/apiaryio/dredd-clean/packages/dredd-transactions
> node ./scripts/prepack.js

lerna info lifecycle dredd-transactions@9.1.0~postpack: dredd-transactions@9.1.0

> dredd-transactions@9.1.0 postpack /Users/kyle/Projects/apiaryio/dredd-clean/packages/dredd-transactions
> node ./scripts/postpack.js

lerna info lifecycle dredd@12.2.0~prepack: dredd@12.2.0

> dredd@12.2.0 prepack /Users/kyle/Projects/apiaryio/dredd-clean/packages/dredd
> npm run build


> dredd@12.2.0 build /Users/kyle/Projects/apiaryio/dredd-clean/packages/dredd
> tsc --build tsconfig.json

lerna http fetch PUT 401 https://registry.npmjs.org/dredd-transactions 1838ms
? This operation requires a one-time password: <redacted>
lerna success published dredd-transactions 9.1.0
lerna notice
lerna notice 📦  dredd-transactions@9.1.0
lerna notice === Tarball Contents ===
lerna notice 1.1kB LICENSE
lerna notice 773B  compile/compileAnnotation.js
lerna notice 1.3kB compile/compileURI/compileParams.js
lerna notice 238B  compile/compileTransactionName.js
lerna notice 2.5kB compile/detectTransactionExampleNumbers.js
lerna notice 2.1kB compile/compileURI/expandURItemplate.js
lerna notice 1.9kB compile/compileURI/index.js
lerna notice 7.6kB compile/index.js
lerna notice 111B  index.js
lerna notice 1.5kB parse/index.js
lerna notice 1.3kB compile/compileURI/validateParams.js
lerna notice 1.7kB package.json
lerna notice 8.1kB README.md
lerna notice === Tarball Details ===
lerna notice name:          dredd-transactions
lerna notice version:       9.1.0
lerna notice filename:      dredd-transactions-9.1.0.tgz
lerna notice package size:  802.8 kB
lerna notice unpacked size: 4.1 MB
lerna notice shasum:        5f51e40af2bf1530f5428109360ff14c99988967
lerna notice integrity:     sha512-3H0IRSzDPjKdi[...]itzwS3vt1hrZg==
lerna notice total files:   40
lerna notice
lerna http fetch PUT 200 https://registry.npmjs.org/dredd-transactions 4072ms
lerna success published dredd 12.2.0
lerna notice
lerna notice 📦  dredd@12.2.0
lerna notice === Tarball Contents ===
lerna notice 444B   bin/dredd
lerna notice 1.1kB  LICENSE
lerna notice 3.4kB  build/addHooks.js
lerna notice 3.4kB  build/annotationToLoggerInfo.js
lerna notice 11.1kB build/reporters/ApiaryReporter.js
lerna notice 4.7kB  build/configuration/applyConfiguration.js
lerna notice 1.7kB  build/configuration/applyLoggingOptions.js
lerna notice 1.8kB  build/reporters/BaseReporter.js
lerna notice 7.9kB  build/childProcess.js
lerna notice 15.2kB build/CLI.js
lerna notice 4.3kB  build/reporters/CLIReporter.js
lerna notice 815B   build/compileTransactionName.js
lerna notice 3.9kB  build/configureReporters.js
lerna notice 1.4kB  build/configUtils.js
lerna notice 2.6kB  build/reporters/DotReporter.js
lerna notice 8.5kB  build/Dredd.js
lerna notice 1.2kB  build/general.js
lerna notice 1.0kB  build/getGoBinary.js
lerna notice 998B   build/getProxySettings.js
lerna notice 2.3kB  build/Hooks.js
lerna notice 716B   build/hooksLog.js
lerna notice 16.7kB build/HooksWorkerClient.js
lerna notice 4.4kB  build/reporters/HTMLReporter.js
lerna notice 560B   build/ignorePipeErrors.js
lerna notice 319B   build/configuration/index.js
lerna notice 542B   build/index.js
lerna notice 12.5kB build/init.js
lerna notice 231B   build/isURL.js
lerna notice 591B   build/logger.js
lerna notice 3.9kB  build/reporters/MarkdownReporter.js
lerna notice 5.1kB  build/configuration/normalizeConfig.js
lerna notice 7.0kB  build/reporters/NyanReporter.js
lerna notice 5.6kB  build/performRequest.js
lerna notice 1.9kB  build/prettifyResponse.js
lerna notice 2.4kB  build/readLocation.js
lerna notice 961B   build/reporters/reporterOutputLogger.js
lerna notice 1.1kB  build/resolveLocations.js
lerna notice 622B   build/resolveModule.js
lerna notice 2.0kB  build/resolvePaths.js
lerna notice 2.0kB  build/sortTransactions.js
lerna notice 28.0kB build/TransactionRunner.js
lerna notice 2.0kB  build/configuration/validateConfig.js
lerna notice 466B   build/which.js
lerna notice 6.3kB  build/reporters/XUnitReporter.js
lerna notice 5.0kB  options.json
lerna notice 2.5kB  package.json
lerna notice === Tarball Details ===
lerna notice name:          dredd
lerna notice version:       12.2.0
lerna notice filename:      dredd-12.2.0.tgz
lerna notice package size:  43.7 kB
lerna notice unpacked size: 191.2 kB
lerna notice shasum:        ed7b339d55687f0e141fe83e48e664748b8e8165
lerna notice integrity:     sha512-KmfuzzSFVVet6[...]aJRURn3TcOgRw==
lerna notice total files:   46
lerna notice
lerna http fetch PUT 200 https://registry.npmjs.org/dredd 7762ms
Successfully published:
 - dredd-transactions@9.1.0
 - dredd@12.2.0
lerna success published 2 packages

$ git tag -a -m dredd@12.2.0 dredd@12.2.0
$ git tag -a -m dredd-transactions@9.1.0 dredd-transactions@9.1.0
$ git push --tags
Enumerating objects: 2, done.
Counting objects: 100% (2/2), done.
Delta compression using up to 4 threads
Compressing objects: 100% (2/2), done.
Writing objects: 100% (2/2), 219 bytes | 219.00 KiB/s, done.
Total 2 (delta 1), reused 0 (delta 0)
remote: Resolving deltas: 100% (1/1), done.
To https://github.com/apiaryio/dredd.git
 * [new tag]         dredd-transactions@9.1.0 -> dredd-transactions@9.1.0
 * [new tag]         dredd@12.2.0 -> dredd@12.2.0
```

**Pull request merged**